cumulus_library_data_metrics/q_ref_target_valid/q_ref_target_valid.jinja

```diff
@@ -7,30 +7,38 @@ CREATE TABLE data_metrics__q_ref_target_valid_{{ src|lower }}_{{ dest|lower }} A
 {% else %}
 
 WITH
-src_status AS {{ utils.extract_status(src) }}
+src_status AS {{ utils.extract_status(src) }},
 
 {%- if is_array %}
-    , flattened AS (
+    flattened AS (
         SELECT id, t.row AS unnested_field
         FROM {{ src }},
             UNNEST({{ field }}) AS t (row)
-    )
+    ),
 {%- set src = 'flattened' %}
 {%- set field = 'unnested_field' %}
 {%- endif %}
 
+grouped_bad_refs AS (
+    SELECT
+        src.id,
+        ARRAY_AGG(src.{{ field }}.reference) AS target
+    FROM {{ src }} AS src
+    LEFT JOIN {{ dest }} AS dest
+    ON SUBSTRING(src.{{ field }}.reference, LENGTH('{{ dest }}/') + 1) = dest.id
+    WHERE
+        src.{{ field }}.reference LIKE '{{ dest }}/%'
```

Comment:
reminder that we're supposed to use regexp() instead of LIKE for performance reasons.

Reply:
Ah yes thank you - filed https://github.com/smart-on-fhir/cumulus-library-data-metrics/issues/38 and I'll do that as a separate thing.